packages/react-test-renderer/src/__tests__/ReactTestRendererAct-test.js

```diff
@@ -0,0 +1,105 @@
+jest.useRealTimers();
+
+let React;
+let ReactTestRenderer;
+let act;
+
+describe('ReactTestRenderer.act()', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactTestRenderer = require('react-test-renderer');
+    act = ReactTestRenderer.act;
+  });
+  it('can use .act() to flush effects', () => {
+    function App(props) {
+      let [ctr, setCtr] = React.useState(0);
+      React.useEffect(() => {
+        props.callback();
+        setCtr(1);
+      });
+      return ctr;
+    }
+    let called = false;
+    let root;
+    act(() => {
+      root = ReactTestRenderer.create(
+        <App
+          callback={() => {
+            called = true;
+          }}
+        />,
+      );
+    });
+
+    expect(called).toBe(true);
+    expect(root.toJSON()).toEqual('1');
+  });
+
+  it("warns if you don't use .act", () => {
+    let setCtr;
+    function App(props) {
+      let [ctr, _setCtr] = React.useState(0);
+      setCtr = _setCtr;
+      return ctr;
+    }
+
+    ReactTestRenderer.create(<App />);
+
+    expect(() => {
+      setCtr(1);
+    }).toWarnDev([
+      'An update to App inside a test was not wrapped in act(...)',
+    ]);
+  });
+
+  describe('async', () => {
+    it('should work with async/await', async () => {
+      function App() {
+        let [ctr, setCtr] = React.useState(0);
+        async function someAsyncFunction() {
+          await null;
+          setCtr(1);
+        }
+        React.useEffect(() => {
+          someAsyncFunction();
+        }, []);
+        return ctr;
+      }
+      let root;
+      ReactTestRenderer.act(() => {
+        root = ReactTestRenderer.create(<App />);
+      });
+      // using this 'workaround' because it looks like this is how
+      // we're ticking over time in the other TestRenderer tests
+      await act(async () => {
+        await Promise.resolve();
+      });
+
+      expect(root.toJSON()).toEqual('1');
+
+      // await ReactTestRenderer.act(async () => {
```

Comment:
What's going on here?

Reply:
there's a difference in behaviour between the TestRenderer and DOM, and I thought it shouldn't. A similar test in ReactDOM works as I'd expected, but I guess it just works that way though. I can make a separate PR showing the difference. deleted this wall of text here.